python/plugins/processing/algs/qgis/ServiceAreaFromLayer.py

```diff
@@ -222,10 +221,16 @@ def processAlgorithm(self, parameters, context, feedback):
         feedback.pushInfo(self.tr('Loading start points...'))
         request = QgsFeatureRequest()
         request.setFlags(request.flags() ^ QgsFeatureRequest.SubsetOfAttributes)
-        features = QgsProcessingUtils.getFeatures(startPoints, context, request)
+        features = source.getFeatures(request)
```

Comment:
You could add a destinationCrs to the request here to match the CRS for layer - then we'd get transparent handling of network/points in different CRS.